draft-romann-mud-constrained.md

```diff
@@ -72,7 +77,103 @@ Building upon the terminology defined in {{!RFC8520}}, this specification introd
 
 TODO. (Remove if there are no additional terms.)
 
-# Exposing a MUD URL using NDP
+# Architecture
+
+Building upon the MUD architecture specified in {{!RFC8520}}, there are two
+main network components relevant for this document:
+The *Thing* that wants to obtain network access via a Router or Switch, and the
+*MUD Manager* that processes MUD URLs, retrieves MUD files from the
+MUD file server, and configures other network components accordingly.
+
+While exposing a MUD URL via the NDP option defined in {{ndp}} works similar to
+using DHCP(v6), with the router forwarding an emitted MUD URL to the MUD
+manager, both the Thing and the MUD manager can play more active roles when
+using CoAP {{!RFC7252}} for exposing and discovering MUD URLs.
+
+A general overview of the MUD architecture adjusted for using CoAP in a
+constrained environment can be seen in {{arch1-fig}}.
+Here, we can see that both the Thing and the MUD manager (as the recipient of
+the MUD URLs) may initiate the MUD discovery process:
+The Thing can contact and register with MUD URL recipients, e.g. by sending a
+CoAP POST request via Multicast, addressing a well-known registration endpoint.
+Conversely, MUD recipients can initiate the discovery process, e.g. by sending a
+COAP GET request to a well-known URI via multicast.
+Therefore, with the additional parameters introduced in this document, both
+Things and MUD managers can play a more active role during the MUD discovery
+process.
+
+TODO: Refine diagram
+
+<!--
+TODO: Fix once plantuml is included in the GitHub Actions Docker image.
```

Comment:
This is probably not going to happen, by the way. I also tried out mermaid, but that didn't work either. I guess we will need to stick to some ASCII art then and/or generate figures separately. But this is probably not that relevant yet.